clang/lib/CodeGen/CGLoopInfo.cpp

```diff
@@ -895,14 +895,27 @@ void LoopInfoStack::push(BasicBlock *Header, clang::ASTContext &Ctx,
     }
 
     if (IntelFPGAII) {
-      unsigned ValueInt = IntelFPGAII->getInterval();
-      if (ValueInt > 0)
-        setSYCLIInterval(ValueInt);
+      unsigned ValueInt = 0;
+      if (auto *E = IntelFPGAII->getIntervalExpr()) {
```

Comment:
What codegen are we doing if there ISN'T an expression?  

Also, const auto should be possible.

Reply:
Is it no codegen if there is no expression?

Reply:
I guess not?  But this should likely be checked in SEMA and asserted here.  Also, can you write a SEMA test for 'no parameters passed'?  That seems like it should be an error case as well.

Reply:
Agree with that, no need to double-check. 
About test - I see that the test has already have cases for no parameters in `boo()` function

Reply:
I see, thanks for pointing that out.